tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py

```diff
@@ -96,3 +98,332 @@ def test_label_swirl(self, xp, scp):
         x = xp.array(x)
         labels, num_features = scp.ndimage.label(x)
         return labels
+
+
+@testing.gpu
+@testing.with_requires('scipy')
+class TestNdimageSum(unittest.TestCase):
+
+    @testing.for_dtypes([cupy.int32, cupy.float32, cupy.float64, cupy.uint32,
+                         cupy.uint64, cupy.ulonglong, cupy.int64, cupy.uint8,
+                         cupy.uint16])
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_sum(self, xp, scp, dtype):
+        image = xp.arange(100, dtype=dtype)
+        label = testing.shaped_random((100,), xp, dtype=xp.int32, scale=3) + 1
+        index = xp.array([1, 2, 3])
+        return scp.ndimage.sum(image, label, index).astype(dtype)
+
+    @testing.for_dtypes([cupy.int32, cupy.float32, cupy.float64, cupy.uint32,
+                         cupy.uint64, cupy.ulonglong, cupy.int64, cupy.uint8,
+                         cupy.uint16])
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_sum_only_input(self, xp, scp, dtype):
+        image = xp.arange(100, dtype=dtype)
+        return scp.ndimage.sum(image).astype(dtype)
+
+    @testing.for_dtypes([cupy.int32, cupy.float32, cupy.float64, cupy.uint32,
+                         cupy.uint64, cupy.ulonglong, cupy.int64, cupy.uint8,
+                         cupy.uint16])
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_sum_no_index(self, xp, scp, dtype):
+        image = xp.arange(100, dtype=dtype)
+        label = testing.shaped_random((100,), xp, dtype=xp.int32, scale=3) + 1
+        return scp.ndimage.sum(image, label).astype(dtype)
+
+    @testing.for_dtypes([cupy.int32, cupy.float32, cupy.float64, cupy.uint32,
+                         cupy.uint64, cupy.ulonglong, cupy.int64, cupy.uint8,
+                         cupy.uint16])
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_sum_scalar_index(self, xp, scp, dtype):
+        image = xp.arange(100, dtype=dtype)
+        label = testing.shaped_random((100,), xp, dtype=xp.int32, scale=3) + 1
+        return scp.ndimage.sum(image, label, 1).astype(dtype)
+
+    @testing.for_dtypes([cupy.complex64, cupy.complex128])
+    def test_ndimage_sum_wrong_dtype(self, dtype):
+        image = cupy.arange(100, dtype=dtype)
+        label = cupy.random.randint(1, 3, dtype=cupy.int32)
```

Comment:
This doesn't draw three.
```suggestion
        label = cupy.random.randint(1, 4, dtype=cupy.int32)
```